src/librepgp/stream-key.cpp

```diff
@@ -353,6 +353,54 @@ transferable_key_add_userid(pgp_transferable_key_t *key, const char *userid)
     return tuid;
 }
 
+bool
+signature_calculate_certification(const pgp_key_pkt_t *   key,
+                                  const pgp_userid_pkt_t *uid,
+                                  pgp_signature_t *       sig,
+                                  const pgp_key_pkt_t *   signer)
+{
+    if (!key || !uid || !sig || !signer) {
+        RNP_LOG("NULL parameter(s)");
+        return false;
+    }
+
+    rng_t rng = {};
+    if (!rng_init(&rng, RNG_SYSTEM)) {
```

Comment:
The plan has been to drop `RNG_SYSTEM` at some point, so I'd rather see if we can pass the `rng_t` down through the various layers instead.

Reply:
@dewyatt What about moving this to the separate PR? Since it would require changing a lot of internal function declarations, as I think we don't want to make it global static variable.

Reply:
@ni4 Sounds good to me